src/ImageSharp/ColorSpaces/Companding/SRgbCompanding.cs

```diff
@@ -84,19 +82,17 @@ public static class SRgbCompanding
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void Expand(Span<Vector4> vectors)
         {
-#if SUPPORTS_RUNTIME_INTRINSICS
             if (Avx2.IsSupported && vectors.Length >= 2)
             {
                 CompandAvx2(vectors, ExpandTable);
 
                 if (Numerics.Modulo2(vectors.Length) != 0)
                 {
                     // Vector4 fits neatly in pairs. Any overlap has to be equal to 1.
-                    Expand(ref MemoryMarshal.GetReference(vectors.Slice(vectors.Length - 1)));
+                    Expand(ref MemoryMarshal.GetReference(vectors[^1..]));
```

Comment:
The old version may be more performant, unfortunately. This is tracked in https://github.com/dotnet/roslyn/issues/43598, and you can see the difference in [this sharplab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88BlABwhgB5gBPAFwFMA+XAWRQAUdBs3bdcAZ3owAlLgC8PKQwDaAPRQA6TQF0A3OUo5cwpq049eSIdNEXJ0uYocNN1ADZQAxhwHKYmgAyHDAA5mwAFrgAtLgoMgYYAL5AA=).

If it's not perf-sensitive, then got with the new code, as it's cleaner to read.
If Roslyn tackles that issue, perf will increase "for free".

Reply:
🤦‍♂️ This is so frustrating. Why introduce an analyser that promotes worse behaviour?